SCons/Util/__init__.py

```diff
@@ -1278,20 +1326,31 @@ def wait_for_process_to_die(pid) -> None:
 
 # From: https://stackoverflow.com/questions/1741972/how-to-use-different-formatters-with-the-same-logging-handler-in-python
 class DispatchingFormatter(Formatter):
+    """Logging formatter which dispatches to various formatters."""
 
     def __init__(self, formatters, default_formatter) -> None:
         self._formatters = formatters
         self._default_formatter = default_formatter
 
     def format(self, record):
-        formatter = self._formatters.get(record.name, self._default_formatter)
+        # Search from record's logger up to its parents:
```

Comment:
How is this different from previous implementation?

Reply:
additional search. it's an extended version from the same stack overflow article.